src/Microsoft.ML.Core/Utilities/DoubleParser.cs

```diff
@@ -70,9 +70,12 @@ public enum Result
             Error = 3
         }
 
-        public static Result Parse(out Single value, string s, int ichMin, int ichLim)
+        public static Result Parse(out Single value, ReadOnlyMemory<char> s)
         {
-            Contracts.Assert(0 <= ichMin && ichMin <= ichLim && ichLim <= Utils.Size(s));
+            int ichMin = 0;
+            int ichLim = s.Length;
```

Comment:
Similar comments as in other file. You cannot perhaps get rid of `ichMin` (though it should be simplified to `ich` or `i`, since it definitely isn't a `min` in any meaningful sense), but bounds checks ought to be against the length of the input `s`.

Once you do that, this assert becomes somewhat pointless, since it simplifies to this.

```csharp
i = 0;
Contracts.Assert(0 <= i && i <= s.Length && s.Length <= s.Length);
```

which obviously is not a helpful assert since its very existence causes more confusion than it addresses.